src/VisualStudio/Core/Def/Implementation/Experimentation/EnhancedColorExperiment.cs

```diff
@@ -0,0 +1,371 @@
+            {
+                _hasTextViewOpened = true;
+
+                _colorApplier = new EnhancedColorApplier(_serviceProvider);
+
+                // Check which experimental flights we are in
+                _inUseEnhancedColorsFlight = _experimentationService?.IsExperimentEnabled(UseEnhancedColorsFlight) ?? false;
+                _inStopEnhancedColorsFlight = _experimentationService?.IsExperimentEnabled(StopEnhancedColorsFlight) ?? false;
+
+                // Do not hook settings changed if we have stopped the experiment. 
+                // We will simply remove the enhanced colors if they are applied.
+                if (!_inStopEnhancedColorsFlight)
+                {
+                    _settingsManager = (ISettingsManager)_serviceProvider.GetService(typeof(SVsSettingsPersistenceManager));
+
+                    // We need to update the theme whenever the Preview Setting changes or the VS Theme changes.
+                    _settingsManager.GetSubset(UseEnhancedColorsSetting).SettingChangedAsync += UseEnhancedColorsSettingChangedAsync;
+                    VSColorTheme.ThemeChanged += VSColorTheme_ThemeChanged;
+                }
+
+                VsTaskLibraryHelper.CreateAndStartTask(VsTaskLibraryHelper.ServiceInstance, VsTaskRunContext.UIThreadIdlePriority, UpdateThemeColors);
+            }
+        }
+
+        public void SubjectBuffersDisconnected(IWpfTextView textView, ConnectionReason reason, Collection<ITextBuffer> subjectBuffers)
+        {
+        }
+
+        private void VSColorTheme_ThemeChanged(ThemeChangedEventArgs e)
+        {
+            AssertIsForeground();
+
+            // Wait until things have settled down from the theme change, since we will potentially be changing theme colors.
+            VsTaskLibraryHelper.CreateAndStartTask(VsTaskLibraryHelper.ServiceInstance, VsTaskRunContext.UIThreadIdlePriority, UpdateThemeColors);
+        }
+
+        private Task UseEnhancedColorsSettingChangedAsync(object sender, PropertyChangedEventArgs args)
+        {
+            AssertIsForeground();
+
+            UpdateThemeColors();
+
+            return Task.CompletedTask;
+        }
+
+        private void UpdateThemeColors()
+        {
+            AssertIsForeground();
+
+            var currentThemeId = GetThemeId();
+
+            // Get the preview feature flag value.
+            var useEnhancedColorsSetting = _settingsManager?.GetValueOrDefault(name: UseEnhancedColorsSetting, defaultValue: 0) ?? 0;
+
+            // useEnhancedColorsSetting
+            //  0 -> use value from flight.
+            //  1 -> always use enhanced colors (unless the kill flight is active).
+            // -1 -> never use enhanced colors.
+            var useFlightValue = useEnhancedColorsSetting == 0;
+            var applyEnhancedColors = (useFlightValue && _inUseEnhancedColorsFlight) || useEnhancedColorsSetting == 1;
+            var removeEnhancedColors = !applyEnhancedColors || _inStopEnhancedColorsFlight;
```

Comment:
It's kinda a mind screw to me that applyEnhancedColors and removeEnhancedColors don't end up just being opposites of each other. And in the case of _inStopEnhancedColorsFlight, we did all of that fancy Boolean logic to ignore it. Can we just write this as:

```
if (_inStopEnhancedColorsFlight || useEnhancedColorsSetting == -1)
{
    _colorApplier.TrySetDefaultColors(currentThemeId);
}
else if (useEnhancedColorsSettings == 1 || (useEnhancedColorsSetting == 0 && _inUseEnhancedColorsFlight))
{
    _colorApplier.TrySetEnhancedColors(currentThemeId);
}
```

In my case it also means if you're not in the flight we just won't do anything at all, rather than potentially resetting colors for fun.